### PR TITLE
Stream transport

### PR DESCRIPTION
**Pull request by runk**

### CODE REVIEW

**Comment by runk** [comment]
Sorry for the amount of changes, but cannot get away without touching every single test file. 😢

File: lib/statsd.js [hunk 1]
```diff
@@ -64,7 +65,7 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
   this.telegraf = options.telegraf || false;
   this.maxBufferSize = options.maxBufferSize || 0;
   this.sampleRate = options.sampleRate || 1;
-  this.bufferFlushInterval = options.bufferFlushInterval || 1000;
+  this.bufferFlushInterval = options.bufferFlushInterval || (process.env.NODE_ENV === 'test' ? 10 : 1000);
```
Comment (runk) [inline lib/statsd.js]: This makes tests to run a lot faster.

File: package.json [hunk 2]
```diff
@@ -30,8 +30,8 @@
   },
   "scripts": {
     "coverage": "nyc --reporter=lcov --reporter=text npm test",
-    "test": "mocha --exit -R spec --timeout 5000 test/*.js",
-    "lint": "eslint lib/**.js test/**.js",
+    "test": "NODE_ENV=test mocha --exit -R spec --timeout 5000 test/*.js",
+    "lint": "eslint lib/**/*.js test/**/*.js",
```
Comment (runk) [inline package.json]: This makes linter to run for all files in all dirs

**Comment by runk** [comment]
@msiebuhr @bdeitte any feedback on the above?

File: lib/statsd.js (hunk 1 above)
Comment (msiebuhr) [reply lib/statsd.js]: Then I think the tests should set it explicitly -- I hate it (as application developer) when underlying libraries changes behaviour between environments...

File: package.json (hunk 2 above)
Comment (msiebuhr) [reply package.json]: I thought `npm test` implied setting `NODE_ENV=test`...

Comment (runk) [reply package.json]: Sorry for misleading comment, I had to go different path: https://github.com/brightcove/hot-shots/pull/130/files#diff-831d41ad895e89c8dffa55c52a8bd1bcR112

File: lib/statsd.js (hunk 1 above)
Comment (runk) [reply lib/statsd.js]: As mentioned - now it's set explicitly in testing/mocking code.

**Comment by runk** [comment]
@bdeitte any thoughts?

**Comment by bdeitte** [comment]
@runk Looks good to me.  I can't merge right now, but I'll do so later today unless I hear any more feedback before then from @msiebuhr  or others.

**Pull request merged**